src/views/RenderView.tsx

```diff
@@ -0,0 +1,358 @@
+
+    const [groundGrid] = useState(() => {
+        const totalWidthMeters = 30;
+        const cellsAcross = 30;
+        const centerLineColor = 0x220044;
+        const gridLineColor = 0x222222;
+        return new THREE.GridHelper(totalWidthMeters, cellsAcross, centerLineColor, gridLineColor);
+    });
+
+    const [axisHelper] = useState(() => {
+        const sizeMeters = 0.1;
+        const helper = new THREE.AxesHelper(sizeMeters);
+        const ViconXAxisColor = new THREE.Color('red');
+        const ViconYAxisColor = new THREE.Color('green');
+        const ViconZAxisColor = new THREE.Color('blue');
+        helper.setColors(ViconXAxisColor, ViconZAxisColor, ViconYAxisColor); //swaps standard THREE y and z colors to emulate Vicon coordinate system
+        helper.scale.x *= -1; //invert x-axis to match Vicon coordinate system
+        return helper;
+    });
+
+    /* An array of un-positioned marker meshes (mapped from the array of marker labels) */
+    const markerMeshes = useMemo(() => {
+        return markerData.markers.map((_,idx) => {
+            const mesh = new THREE.Mesh(new THREE.SphereGeometry(0.01,16,16), new THREE.MeshBasicMaterial());
+            mesh.visible = false;
+            mesh.userData.index = idx; //make all meshes independently aware of their index within markerMeshes array (for raycast id)
+            return mesh;
+        });
+    }, [markerData.markers]);
+
+    /* An array of two un-positioned force vectors */
+    const [forceVectors] = useState(() => {
+        const vec1 = new THREE.ArrowHelper();
+        const vec2 = new THREE.ArrowHelper();
+        vec1.setColor(0xFF0000);
+        vec2.setColor(0xFF0000);
+        vec1.visible = false;
+        vec2.visible = false;
+        return [vec1, vec2];
+    });
+
+    const [raycaster] = useState(() => new THREE.Raycaster());
+
+    const handleRaycast = useCallback(({
+        clientX, clientY, ctrlKey, shiftKey, currentTarget: {offsetLeft, offsetTop, offsetWidth, offsetHeight}
+    }: React.MouseEvent<HTMLDivElement,MouseEvent>) => {
+        /* Determine type of selection */
+        const multiSelect = ctrlKey || shiftKey;
+        /* Raycaster requires coordinates to be transformed to range [-1,1], with positive y toward top of page (inverted from CSS) */
+        const percentRight = (clientX-offsetLeft)/offsetWidth;
+        const percentDown = (clientY-offsetTop)/offsetHeight;
+        const percentUp = 1-percentDown;
+        const coords = {
+            x: 2*percentRight - 1,
+            y: 2*percentUp - 1,
+        };
+        /* Perform raycast */
+        raycaster.setFromCamera(coords, camera);
+        const hitList = raycaster.intersectObjects(markerMeshes, true);
+        const hit: number|undefined = hitList[0]?.object.userData.index; //get idx of the closest mesh to the camera, if present
+        /* Update selected markers list with result  */
+        setSelectedMarkers.call(undefined, currentList =>
+            hit||hit===0?
```

Comment:
what

Reply:
I get a little mixed up sometimes dealing with the whole truthy-falsy thing.
I presume the preferable condition would be `hit !== undefined`?
Fixed.